pymodbus/repl/client/main.py

```diff
@@ -1,25 +1,9 @@
 """Pymodbus REPL Entry point."""
-# pylint: disable=anomalous-backslash-in-string
-# flake8: noqa
 import logging
 import pathlib
-import sys
-
-
-try:
-    import click
-except ImportError:
-    print('click not installed!! Install with "pip install click"')
-    sys.exit(1)
-try:
-    from prompt_toolkit import PromptSession, print_formatted_text
-except ImportError:
-    print(
-        "prompt toolkit is not installed!! "
-        'Install with "pip install prompt_toolkit --upgrade"'
-    )
-    sys.exit(1)
 
+import click
```

Comment:
This will not work !!! the import needs to be behind a try/except, otherwise you require click to be installed always (you will see the same in  client/serial and other places).

Reply:
The import is only needed for REPL, which can't be used anyway without `click`.  the existing code which I removed calls `sys.exit(1)`

https://github.com/pymodbus-dev/pymodbus/blob/d2c7c778707d8e0cc1ea9ea30ffccf945ea56b89/pymodbus/repl/client/main.py#L9-L13

Reply:
Yes but we have had problems, latest with sqlalchemy because python loads the files and thus tries to do the import. The problem arises is the package is referenced (__jnit__) so for optional libraries we prefer to use try/except.

Having the try/except allows python to pass the module without having click installed (which is optional).

Reply:
This is the "modern" way we handle optional packages:
```
try:
    import serial
except ImportError:
    pass
```

Reply:
https://github.com/pymodbus-dev/pymodbus/pull/1114 was the import error with `pyserial`, which was caused because it was imported in the generic server class.  This is called by the normal server/client code paths.
https://github.com/pymodbus-dev/pymodbus/issues/1319 was the import error with `sqlalchemy` and `redis`, which was caused because they were imported in the datastore.  This is called by the normal server/client code paths.

I don't think this is true here, since no other code in the normal server/client code paths calls the REPL.  If something did, it would already trigger the (existing) `sys.exit(1)`.  I have tested using the pymodbus client without `click` installed, and it's fine.  So the Try/Except is not protecting anything.

As far as I can tell, if `click` is not installed the REPL is useless.  So the Try/Except only changes the `ImportError` to a `NameError`.

I can add the Try/Except, of course, but don't think it has any use.

Reply:
Your analysis is correct, I was merely preparing for the future where __init__ contain all external classes, and deep links are not allowed.

Lets leave it for now.